app/controllers/api/custom_buttons_controller.rb

```diff
@@ -8,6 +8,8 @@ def create_resource(_type, _id, data)
         custom_button.options = data["options"].deep_symbolize_keys if data["options"]
         custom_button.save!
         custom_button.create_resource_action!(data["resource_action"].deep_symbolize_keys) if data.key?("resource_action")
+        data.except!('resource_action')
+        custom_button.update!(data.deep_symbolize_keys) if data.present?
```

Comment:
Would this remove the need for https://github.com/ManageIQ/manageiq-api/pull/955/files#diff-589b6e24c5d5b0911bc19b4576b07fbe419a19f5fb83b768939331a740f3dd89R8 ?

Reply:
and https://github.com/ManageIQ/manageiq-api/pull/955/files#diff-589b6e24c5d5b0911bc19b4576b07fbe419a19f5fb83b768939331a740f3dd89R5